query/stdlib/influxdata/influxdb/metrics.go

```diff
@@ -0,0 +1,87 @@
+package influxdb
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/flux/execute"
+	platform "github.com/influxdata/influxdb"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	orgLabel = "org"
+	opLabel  = "op"
+)
+
+type metrics struct {
+	ctxLabelKeys []string
+	requestDur   *prometheus.HistogramVec
+}
+
+// NewMetrics produces a new metrics objects for an influxdb source.
+// Currently it just collects the duration of read requests into a histogram.
+// ctxLabelKeys is a list of labels to add to the produced metrics.
+// The value for a given key will be read off the context.
+// The context value must be a string or an implementation of the Stringer interface.
+// In addition, produced metrics will be labeled with the orgID and type of operation requested.
+func NewMetrics(ctxLabelKeys []string) *metrics {
+	labelKeys := make([]string, len(ctxLabelKeys)+2)
+	copy(labelKeys, ctxLabelKeys)
+	labelKeys[len(labelKeys)-2] = orgLabel
+	labelKeys[len(labelKeys)-1] = opLabel
+
+	m := new(metrics)
+	m.requestDur = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "query",
+		Subsystem: "influxdb_source",
+		Name:      "read_request_duration_seconds",
+		Help:      "Histogram of times spent in read requests",
+		Buckets:   prometheus.ExponentialBuckets(1e-3, 5, 7),
+	}, labelKeys)
+	m.ctxLabelKeys = ctxLabelKeys
+	return m
+}
+
+// PrometheusCollectors satisfies the PrometheusCollector interface.
+func (m *metrics) PrometheusCollectors() []prometheus.Collector {
+	if m == nil {
+		return nil
+	}
```

Comment:
Why is this check for `nil` necessary 🤔 ?

Reply:
Well... without this check if I just point the current idpe master at this, it will crash in the pipeline tests because `m` is `nil`.  I wanted this set of changes to be okay without any changes to idpe.  Do you think it's okay?  In any case I'll add a comment.